corehq/apps/userreports/models.py

```diff
@@ -205,6 +205,73 @@ def is_rebuilding_in_place(self):
     def is_rebuild_in_progress(self):
         return self.is_rebuilding or self.is_rebuilding_in_place
 
+    def rebuild_failed(self, data_source_config_id):
+        """
+        Returns ``True`` if the rebuild failed, ``False`` if it succeeded
+        or has not yet failed, or ``None`` if Flower is not available.
+        """
+        flower_url = getattr(settings, 'CELERY_FLOWER_URL')
+
+        def none_max(a, b):
+            if a is None:
+                return b
+            if b is None:
+                return a
+            return max(a, b)
+
+        def format_datetime(dt):
+            return dt.strftime('%Y-%m-%d %H:%M')
+
+        def is_this_data_source(rebuild_task):
+            args = literal_eval(rebuild_task['args'])  # a tuple
+            return args[0] == data_source_config_id
+
+        def iter_tasks():
+            task_names = (
+                'corehq.apps.userreports.tasks.rebuild_indicators',
+                'corehq.apps.userreports.tasks.rebuild_indicators_in_place',
+                'corehq.apps.userreports.tasks.resume_building_indicators',
+            )
+            initiated_at = none_max(self.initiated, self.initiated_in_place)
+            start = format_datetime(initiated_at - timedelta(seconds=60))
+            for task_name in task_names:
+                tasks = requests.get(
+                    flower_url + '/api/tasks',
+                    params={
+                        'taskname': task_name,
+                        'received_start': start,
+                    },
+                    timeout=3,
+                ).json()
+                for task_uuid, task in tasks.items():
+                    if is_this_data_source(task):
+                        yield task
+
+        if not self.initiated and not self.initiated_in_place:
+            # The rebuild task hasn't started.
+            return False
+
+        if (
+            (self.initiated and self.finished)
+            or (self.initiated_in_place and self.finished_in_place)
+        ):
+            # The rebuild task completed.
+            return False
+
+        if not flower_url:
+            # We are unable to find out about the rebuild task.
+            return None
+
+        rebuild_tasks = sorted(iter_tasks(), key=lambda t: t['started'])
+        if rebuild_tasks:
+            # Return True if the last task failed, otherwise return False.
+            # See https://docs.celeryq.dev/en/stable/reference/celery.result.html#celery.result.AsyncResult.state
+            return rebuild_tasks[-1]['state'] == 'FAILURE'
```

Comment:
It would be nice to have a place where we define Celery constants like this, but not necessary for this PR I think.

Reply:
Good call! Celery defines these constants: https://github.com/dimagi/commcare-hq/pull/33063/commits/cced68ef86177069fdb788b17ae3b47dc64752b5